src/HexagonOptimize.cpp

```diff
@@ -914,6 +914,9 @@ class OptimizePatterns : public IRMutator {
     }
 
     Expr visit(const Call *op) override {
+        if (op->is_intrinsic(Call::if_then_else)) {
```

Comment:
I think you should only skip this if the condition is a vector. If the condition is a scalar it should be fine and I think might be an important case. Same for the other 4 instances.

Reply:
Done

Reply:
This is different from what I suggested. This skips it for vector *results*, which are common. The case I am thinking about that we should still match patterns for is if_then_else where the *condition* (the first argument of the call) is a scalar, but the result and operands are vectors.

Reply:
Oh understood now. I'm assuming we should pattern match where conditions are Broadcasts as well?

Reply:
I think so. I think we might see if_then_else with both scalar or broadcasting conditions, I'd have to poke around in the code to be sure one way or the other.

Reply:
Yeah. It seems like we can encounter both. Added condition for Broadcasts.